src/JustSaying/JustSayingBus.cs

```diff
@@ -46,15 +49,9 @@ public IMessageMonitor Monitor
         public IMessageSerializationRegister SerializationRegister { get; }
         public IMessageBackoffStrategy MessageBackoffStrategy { get; set; }
 
-        public IMessageLockAsync MessageLock
-        {
-            get => HandlerMap.MessageLock;
-            set => HandlerMap.MessageLock = value;
-        }
-
         public IMessageContextAccessor MessageContextAccessor { get; set; }
 
-        public HandlerMap HandlerMap { get; }
+        public MiddlewareMap MiddlewareMap { get; }
```

Comment:
If we hid this from the public API we could make `MiddlewareMap` internal (we'd need to also make `MessageDispatcher` internal, which is probably a good idea).
Perhaps we evaluate what we can hide as a separate task outside of this PR.

Reply:
We can make this internal while leaving the MessageDispatcher as is - MiddlewareMap doesn't have the same strange dependencies that HandlerMap had because it doesn't explicitly know about message locks anymore.

I think its still a good idea to review visibility separately - we might be able to hide some more things now that we couldn't before 👍